hs_core/views/resource_access_api.py

```diff
@@ -0,0 +1,182 @@
+
+from rest_framework.response import Response
+from rest_framework import generics, serializers
+from rest_framework import status
+
+from hs_core import hydroshare
+from hs_core.hydroshare import utils
+from hs_access_control.models import UserResourcePrivilege, GroupResourcePrivilege, \
+    PrivilegeCodes, UserAccess
+from hs_core.views import utils as view_utils
+from hs_core.views.utils import ACTION_TO_AUTHORIZE
+
+
+class PrivilegeField(serializers.Field):
+    def to_representation(self, privilege):
+        return PrivilegeCodes.PRIVILEGE_CHOICES[privilege-1][1]
+
+
+class GroupResourcePrivilegeSerializer(serializers.ModelSerializer):
+    privilege = PrivilegeField()
+
+    class Meta:
+        model = GroupResourcePrivilege
+        fields = ('id', 'privilege', 'group', 'resource', 'grantor')
+
+
+class UserResourcePrivilegeSerializer(serializers.ModelSerializer):
+    privilege = PrivilegeField()
+
+    class Meta:
+        model = UserResourcePrivilege
+        fields = ('id', 'privilege', 'user', 'resource', 'grantor')
+
+
+class ResourceAccessUpdateDelete(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Read, update, or delete a resource
+
+    REST URL: hsapi/resource/{pk}/access
+    HTTP method: GET
+    :return: (on success): JSON representation of resource access with 'groups' and 'users' keys.
+
+    REST URL: hsapi/resource/{pk}/access?(user_id=#|group_id=#)
+    HTTP method: DELETE
+
+    :type str
+    :param user_id: user ID to remove
+    :type str
+    :param group_id: group ID to remove
+    :return: (on success): Success or Error JSON object
+
+    REST URL: hsapi/resource/{pk}/access
+    HTTP method: PUT
+    :return: (on success): Success or Error JSON object
+
+    :type str
+    :param user_id: user ID to remove
+    :type str
+    :param group_id: group ID to remove
+    :type PrivilegeCode int
+    :param privilege: PrivilegeCode to specifiy access level
+    :return: (on success): Success or Error JSON objectit
+    """
+
+    serializer_class = UserResourcePrivilegeSerializer
+
+    allowed_methods = ('GET', 'PUT', 'DELETE')
+
+    def get(self, request, pk):
+        view_utils.authorize(request, pk,
+                             needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE_ACCESS)
+
+        user_resource_serializer, group_resource_serializer = self.get_serializer_classes()
+        user_resource_privilege, group_resource_privilege = self.get_queryset(pk, request.user)
+
+        response_data = dict()
+        response_data['users'] = \
+            user_resource_serializer(user_resource_privilege, many=True).data
+        response_data['groups'] = \
+            group_resource_serializer(group_resource_privilege, many=True).data
+
+        return Response(data=response_data, status=status.HTTP_200_OK)
+
+    def put(self, request, pk):
+        view_utils.authorize(request, pk,
```

Comment:
The permission needed for the resource depends on the 'access privilege' the user trying to grant someone. Obviously the minimum permission needed is "VIEW_RESOURCE_ACCESS". @alvacouch should we check for "ACTION_TO_AUTHORIZE.VIEW_RESOURCE_ACCESS"? for all access related http methods?